src/components/RadioTile/RadioTile.tsx

```diff
@@ -0,0 +1,94 @@
+import * as RadixRadioGroup from "@radix-ui/react-radio-group";
+import { Box, sprinkles, Text } from "@saleor/macaw-ui-next";
+import React from "react";
+
+import { getBgColor, getBorderColor, getHoverState } from "./utils";
+
+export interface RadioTileProps {
+  checked: boolean;
+  title: string;
+  description: string;
+  value: string;
+}
+
+export const RadioTile = ({
+  checked,
+  title,
+  description,
+  value,
+}: RadioTileProps) => {
+  const [isHoverState, setHoverState] = React.useState(false);
+
+  return (
+    <RadixRadioGroup.Item value={value} asChild>
+      <Box
+        position="relative"
+        as="label"
+        gap={2}
+        display="grid"
+        __gridTemplateColumns="auto 1fr"
+        alignItems="center"
+        borderColor={getBorderColor({ checked, isHoverState })}
+        borderWidth={1}
+        borderStyle="solid"
+        borderRadius={2}
+        padding={3}
+        cursor="pointer"
+        onMouseEnter={() => setHoverState(true)}
+        onMouseLeave={() => setHoverState(false)}
+      >
+        <Box
+          width={5}
+          height={5}
+          borderRadius="100%"
+          display="flex"
+          justifyContent="center"
+          alignItems="center"
+          backgroundColor={getHoverState({ checked, isHoverState })}
```

Comment:
suggestion: I'd name function `getHoverState` in the way that suggest what it does